### ISSUE TITLE
Alpine package version pinning rule too strict?

### ISSUE THREAD

**Issue by jonasholtkamp** [hadolint/hadolint#204]
- [ ] This is a bug report
- [ ] This is a feature request
- [x] I searched existing issues before opening this one

### Context
#134 introduced the feature for linting Alpine Linux package version pinning. However, the Alpine package manager strategy apparently is to don't keep old package versions in a particular Alpine version. This means, when a new package version for a particular Alpine Linux version is published, the older ones won't be available anymore.

Example: In [Alpine 3.6](https://pkgs.alpinelinux.org/packages?name=git&branch=v3.6&repo=main) the only available git version is *2.13.7-r0*. The previously in 3.6 available version *2.13.5-r0* is no longer available then. This to me is not compatible with Dockerfile linting with hadolint. One option would be to deactivate the rule DL3018, or ... read on:

### Expected behavior
Since `docker build` with an *exact* pinning (e.g. `git=2.13.5-r0`) fails after a new git version is published to the configured repo (remember, old versions are deleted), I expected to pass the linting with a "semi-pinned" version like `git<2.14`.  This would avoid altering the Dockerfile everytime a new git version is published for Alpine Linux.

### Actual behavior
"Semi-pinning" a version fails because of rule DL3018:
```bash
$ docker run --rm -i hadolint/hadolint < Dockerfile
/dev/stdin:2 DL3018 Pin versions in apk add. Instead of `apk add <package>` use `apk add <package>=<version>`
```
### Steps to reproduce the behavior
```bash
$ echo  "FROM alpine:3.6

RUN apk add --no-cache git=2.13.5-r0" > Dockerfile
$ docker build .
```
Result should be:
```bash
ERROR: unsatisfiable constraints:
  git-2.13.7-r0:
    breaks: world[git=2.13.5-r0]
The command '/bin/sh -c apk add --no-cache git=2.13.5-r0' returned a non-zero code: 1
```

The only way to fix this without dropping the rule DL3018 would be to bump the version to `git-2.13.7-r0`. However, in a matter of days or weeks this will lead to an build error, because the old version won't be available any longer after a new version has been published.

**Comment by lorenzo** [comment]
@jonasholtkamp does `git=2.13.*` work in alpine like it would work with `apt-get`?

**Comment by jonasholtkamp** [comment]
@lorenzo, no that doesn't work:
```bash
docker build .
Sending build context to Docker daemon  2.048kB
Step 1/2 : FROM alpine:3.6
 ---> 77144d8c6bdc
Step 2/2 : RUN apk add --no-cache git=2.13.*
 ---> Running in 6f655f9112de
fetch http://dl-cdn.alpinelinux.org/alpine/v3.6/main/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.6/community/x86_64/APKINDEX.tar.gz
ERROR: 'git=2.13.*' is not a valid dependency, format is name(@tag)([<>=]version)
```

**Comment by lorenzo** [comment]
Thanks for checking, I will investigate if we there are other ways to do this

**Comment by lorenzo** [comment]
There is this way, it may work https://www.google.dk/amp/s/til.codes/how-to-install-a-specific-package-version-in-alpine-and-docker/amp/

**Comment by jonasholtkamp** [comment]
Thanks @zemanlx, this is awesome! It's perfectly fine for us to use `alpine:3.7`. I just tested your code snippet both with `docker build .` and `docker run --rm -i hadolint/hadolint < Dockerfile` and it works just fine. I think this resolves the issue by itself :)

**Comment by zemanlx** [comment]
@jonasholtkamp I am glad that It helps. Alpine docs are spartan, but this is the best I was able to find 

>This is useful for requirements such as: python3=~3.6, which would match python3-3.6.[0-9].
This implementation should in theory be backwards compatible with the implementation in Adelie. 

Details [here](https://git.alpinelinux.org/cgit/apk-tools/commit/?id=693b4bcdb0f22904a521a7c8ac4f13e697dc4d71).
